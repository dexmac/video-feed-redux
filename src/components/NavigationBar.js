import React from "react";
import { Navbar } from "react-bootstrap";
import PropTypes from "prop-types";

const NavigationBar = ({ brand = "" }) => (
  <Navbar>
    <Navbar.Header>
      <Navbar.Brand>
        <div className="navbar-brand">{brand}</div>
      </Navbar.Brand>
    </Navbar.Header>
  </Navbar>
);

NavigationBar.propTypes = {
  brand: PropTypes.string
};

export default NavigationBar;
