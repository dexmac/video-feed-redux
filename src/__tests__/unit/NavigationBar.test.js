import React from "react";
import NavigationBar from "../../components/NavigationBar";

describe("NavigationBar", () => {
  let props;
  let mountedNavBar;

  /**
   * Mounts the component using Enzyme if it is not mounted yet
   * @returns {object} - returns the mounted component
   */
  const navBar = () => {
    if (!mountedNavBar) {
      mountedNavBar = mount(<NavigationBar {...props} />);
    }
    return mountedNavBar;
  };

  /**
   * Clearing props before each test
   */
  beforeEach(() => {
    props = {
      brand: undefined
    };
    mountedNavBar = undefined;
  });

  /**
   * Shallow rendering vs. snapshot test
   */
  it("renders correctly && matches snapshot", () => {
    const tree = shallow(<NavigationBar />);
    expect(tree).toMatchSnapshot();
  });

  describe("when `brand` is defined", () => {
    beforeEach(() => {
      props.brand = "mybrand";
    });

    it("sets the rendered `div` text to the same value as `brand`", () => {
      const myDiv = navBar().find("div.navbar-brand");
      expect(myDiv.text()).toBe(props.brand);
    });
  });

  describe("when `brand` is undefined", () => {
    beforeEach(() => {
      props.brand = undefined;
    });

    it("sets the rendered `div` text to be an empty string", () => {
      const myDiv = navBar().find("div.navbar-brand");
      expect(myDiv.text()).toBe("");
    });
  });
});
