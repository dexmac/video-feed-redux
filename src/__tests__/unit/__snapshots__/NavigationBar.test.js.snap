// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly && matches snapshot 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
>
  <NavbarHeader>
    <NavbarBrand>
      <div
        className="navbar-brand"
      />
    </NavbarBrand>
  </NavbarHeader>
</Uncontrolled(Navbar)>
`;
